lib/redix.ex

```diff
@@ -780,22 +780,29 @@ defmodule Redix do
 
     telemetry_metadata = %{
       connection: conn,
-      commands: commands,
-      start_time: System.system_time()
+      commands: commands
     }
 
     start_time = System.monotonic_time()
+    :telemetry.execute([:redix, :pipeline, :start], %{system_time: System.system_time()}, telemetry_metadata)
 
     case Redix.Connection.pipeline(conn, commands, timeout) do
       {:ok, response} ->
         end_time = System.monotonic_time()
-        measurements = %{elapsed_time: end_time - start_time}
-        :ok = :telemetry.execute([:redix, :pipeline], measurements, telemetry_metadata)
+        measurements = %{duration: end_time - start_time}
+        :ok = :telemetry.execute([:redix, :pipeline, :stop], measurements, telemetry_metadata)
         {:ok, response}
 
       {:error, reason} ->
-        telemetry_metadata = Map.put(telemetry_metadata, :reason, reason)
-        :ok = :telemetry.execute([:redix, :pipeline, :error], %{}, telemetry_metadata)
+        end_time = System.monotonic_time()
+        measurements = %{duration: end_time - start_time}
+        telemetry_metadata =
+          telemetry_metadata
+          |> Map.put(:reason, reason)
+          |> Map.put(:kind, :error)
+          |> Map.put(:stacktrace, __STACKTRACE__)
+
+        :ok = :telemetry.execute([:redix, :pipeline, :failure], measurements, telemetry_metadata)
```

Comment:
Just so we don't miss it in the shuffle, we have updated the recommendation here to be `[:redix, :pipeline, :stop]`, and I think you wouldn't want a `__STACKTRACE__` there, but you could still put `kind: :error` if you want to signal that the `:stop` event was unsuccessful. Unfortunately, we don't have a convention for that yet, but I think we should come up with one eventually to simplify integrations with other Telemetry collectors.

The `:failure` event has been renamed to `:exception`, to more clearly indicate that it's intended for things that actually crashed in some way and should have a `__STACKTRACE__` attached. If there's anything inside `Redix.Connection.pipeline` that _might_ crash, I'd suggest wrapping this somehow in a `try ... else` so that if something does raise, you could still fire the `:exception` event because we need to be sure that for every `:start` there is either a matching `:stop` or a matching `:exception` so that distributed tracing collectors are able to stack in sync with their span context.

Reply:
Does it make sense to provide both `reason` and `kind` for errors? That way tracers or other handlers can encode that information in some meaningful way?